backend/app.py

```diff
@@ -1094,5 +1097,32 @@ def get_product_prices_endpoint():
         logging.error(f"Failed to retrieve prices: {e}")
         return jsonify({"error": str(e)}), 500
 
+@app.route("/subscription/<subscriptionId>/financialAssistant", methods=["PUT"])
+def financial_assistant(subscriptionId):
+    client_principal_id = request.headers.get("X-MS-CLIENT-PRINCIPAL-ID")
```

Comment:
@vmlcode before this you might want to add a proper valdation of the subscription Id:
```
        # Validate subscription_id
        if not subscription_id or not isinstance(subscription_id, str):
            raise BadRequest("Invalid subscription ID")
```

and you going to need to add a error handle for this if it fails